solidity/ecdsa/contracts/libraries/Wallets.sol

```diff
@@ -67,6 +67,18 @@ library Wallets {
         self.registry[walletID].publicKeyY = publicKeyY;
     }
 
+    /// @notice Deletes wallet with the given ID from the registry. Reverts
+    ///         if wallet with the given ID has not been registered or if it
+    ///         has already been closed.
+    function deleteWallet(Data storage self, bytes32 walletID) internal {
+        require(
+            isWalletRegistered(self, walletID),
+            "Wallet with the given ID has not been registered"
+        );
+
+        delete self.registry[walletID];
```

Comment:
I'm wondering whether deleting is enough here. What if someone submits a deleted wallet pubkey via `approveDkgResult` and registers it again? I know the result must be signed by multiple members but still. Shouldn't we store an additional piece of info that will prevent re-adding an already deleted wallet?

Reply:
Yep that sounds good!